src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,103 @@
-import { useState } from 'react';
+/* eslint-disable max-len */
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+
+  function toTrim(string: string) {
+    return string.trim();
+  }
+
+  const [count, setCount] = useState(0);
+
+  const [newMovie, setNewMovie] = useState<Movie>({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const isDisabled = !toTrim(newMovie.title) || !toTrim(newMovie.imgUrl) || !toTrim(newMovie.imdbUrl) || !toTrim(newMovie.imdbId);
+
+  const reset = () => {
+    setNewMovie({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+    setCount(count + 1);
+  };
+
+  const handleChange = (name: string, value: string) => {
+    setNewMovie({ ...newMovie, [name]: value });
+  };
```

Comment:
```suggestion
  const handleChange = (event: React.ChangeEvent<HtmlInputElement>) => {
    const { value, name } = event.target;
    setNewMovie({ ...newMovie, [name]: value });
  };
```
use event.target.name and pass just reference on this handler in TextField